### PR TITLE
fix some racist option names

### PR DESCRIPTION
**Pull request by jsteemann**
### Scope & Purpose

Fix some racist startup option and variable names.
This is still work in progress.

- [x] Bug-Fix for *devel-branch* (i.e. no need for backports?)
- [x] The behavior in this PR can be (and was) *manually tested* (support / qa / customers can test it)
- [x] The behavior change can be verified via automatic tests

### Testing & Verification

This change is already covered by existing tests, such as *permissions, server_permissions*.

### CODE REVIEW

**Comment by pluma** [comment]
While we're addressing toxicity, it might be worth to take a look at the [self-defined dictionary](https://www.selfdefined.app) which among other things list examples of toxic language and ways to replace them, e.g. https://www.selfdefined.app/definitions/sanity-check/

**Comment by jsteemann** [comment]
http://172.16.10.101:8080/view/PR/job/arangodb-matrix-pr/10624/

**Comment by jsteemann** [comment]
@pluma: just replaced "sanity check" in many places. I won't have time to scan the whole dictionary. 
Let me know of other terms that you think are inadequate. 
Thanks!

**Comment by pluma** [comment]
At a glance we seem to be clear on most of the relevant ones, except for some 3rdParty code that is not under our control. :+1:

**Comment by jsteemann** [comment]
@pluma: option names seem to be ok now. The internal code still has a lot of occurrences of the terms "master" and "slave", for the replication. I think I will address this in a separate PR, otherwise this PR will get too big.

File: arangod/Agency/State.h [hunk 1]
```diff
@@ -238,7 +238,7 @@ class State {
   /// @brief Check collections
   bool checkCollections();
 
-  /// @brief Check collection sanity
+  /// @brief Check collection existence
```
Comment (dothebart) [inline arangod/Agency/State.h:241]: /// @brief Check collection integrity

File: arangod/Aql/QueryList.h [hunk 2]
```diff
@@ -118,7 +118,7 @@ class QueryList {
   /// modifications of this variable are possible but are considered unharmful
   inline void slowQueryThreshold(double value) {
     if (value < 0.0 || value == HUGE_VAL || value != value) {
-      // sanity checks
+      // basic checks
```
Comment (dothebart) [inline arangod/Aql/QueryList.h:121]: do these comments have a value afterall? maybe simply delete them?

File: arangod/Cluster/ClusterMethods.cpp [hunk 3]
```diff
@@ -2317,7 +2317,7 @@ std::vector<std::shared_ptr<LogicalCollection>> ClusterMethods::createCollection
     bool waitForSyncReplication, bool enforceReplicationFactor,
     bool isNewDatabase, std::shared_ptr<LogicalCollection> const& colToDistributeShardsLike) {
   TRI_ASSERT(parameters.isArray());
-  // Collections are temporary collections object that undergoes sanity checks
+  // Collections are temporary collections object that undergoes quick checks
```
Comment (dothebart) [inline arangod/Cluster/ClusterMethods.cpp:2320]: integrity

File: arangod/IResearch/IResearchAnalyzerFeature.cpp [hunk 4]
```diff
@@ -2688,7 +2688,7 @@ void IResearchAnalyzerFeature::start() {
   }
 
 #ifdef ARANGODB_ENABLE_MAINTAINER_MODE
-  // sanity check: we rely on this condition is true internally
+  // basic check: we rely on this condition is true internally
```
Comment (dothebart) [inline arangod/IResearch/IResearchAnalyzerFeature.cpp:2691]: integrity

File: arangod/RestServer/DatabaseFeature.cpp [hunk 5]
```diff
@@ -329,7 +329,7 @@ void DatabaseFeature::collectOptions(std::shared_ptr<ProgramOptions> options) {
 }
 
 void DatabaseFeature::validateOptions(std::shared_ptr<ProgramOptions> options) {
-  // sanity check
+  // basics checks...
```
Comment (dothebart) [inline arangod/RestServer/DatabaseFeature.cpp:332]: integrity

File: arangod/RocksDBEngine/RocksDBTransactionState.cpp [hunk 6]
```diff
@@ -300,7 +300,7 @@ arangodb::Result RocksDBTransactionState::internalCommit() {
   }
 
 #ifdef ARANGODB_ENABLE_MAINTAINER_MODE
-  // sanity check our on-disk WAL format
+  // check our on-disk WAL format
```
Comment (dothebart) [inline arangod/RocksDBEngine/RocksDBTransactionState.cpp:303]: integrity

File: arangod/Utils/SingleCollectionTransaction.cpp [hunk 7]
```diff
@@ -99,8 +99,7 @@ LogicalCollection* SingleCollectionTransaction::documentCollection() {
   
 TRI_voc_cid_t SingleCollectionTransaction::addCollectionAtRuntime(std::string const& name,
                                                                   AccessMode::Type type) {
-
-  // sanity check
+  // basic checks
```
Comment (dothebart) [inline arangod/Utils/SingleCollectionTransaction.cpp:102]: delete

File: arangod/V8Server/V8DealerFeature.cpp [hunk 8]
```diff
@@ -522,7 +522,7 @@ void V8DealerFeature::copyInstallationFiles() {
   }
 
   if (overwriteCopy) {
-    // sanity check before removing an existing directory:
+    // basic checks before removing an existing directory:
```
Comment (dothebart) [inline arangod/V8Server/V8DealerFeature.cpp:525]: integrity

File: arangod/V8Server/v8-voccursor.cpp [hunk 9]
```diff
@@ -181,7 +181,7 @@ struct V8Cursor final {
         _cursorId(cursorId),
         _resolver(vocbase),
         _cte(transaction::Context::createCustomTypeHandler(vocbase, _resolver)) {
-    // sanity checks
+    // basic checks
```
Comment (dothebart) [inline arangod/V8Server/v8-voccursor.cpp:184]: remove

File: arangod/VocBase/LogicalCollection.cpp [hunk 10]
```diff
@@ -884,7 +884,7 @@ arangodb::Result LogicalCollection::properties(velocypack::Slice const& slice, b
       }
 
       if (ServerState::instance()->isCoordinator() &&
-          replicationFactor != _sharding->replicationFactor()) {  // sanity checks
+          replicationFactor != _sharding->replicationFactor()) {  // basics checks
```
Comment (dothebart) [inline arangod/VocBase/LogicalCollection.cpp:887]: range

File: arangod/VocBase/LogicalCollection.cpp [hunk 11]
```diff
@@ -937,7 +937,7 @@ arangodb::Result LogicalCollection::properties(velocypack::Slice const& slice, b
       }
 
       if (ServerState::instance()->isCoordinator() &&
-          replicationFactor != _sharding->writeConcern()) {  // sanity checks
+          replicationFactor != _sharding->writeConcern()) {  // basic checks
```
Comment (dothebart) [inline arangod/VocBase/LogicalCollection.cpp:940]: range

File: lib/Basics/VelocyPackHelper.cpp [hunk 12]
```diff
@@ -510,7 +510,7 @@ VPackBuilder VelocyPackHelper::velocyPackFromFile(std::string const& path) {
 
 static bool PrintVelocyPack(int fd, VPackSlice const& slice, bool appendNewline) {
   if (slice.isNone()) {
-    // sanity check
+    // basic check...
```
Comment (dothebart) [inline lib/Basics/VelocyPackHelper.cpp:513]: remove

File: lib/V8/v8-buffer.cpp [hunk 13]
```diff
@@ -1538,7 +1538,7 @@ static void MapSetIndexedBuffer(uint32_t idx, v8::Local<v8::Value> value,
 void TRI_InitV8Buffer(v8::Isolate* isolate) {
   v8::HandleScope scope(isolate);
 
-  // sanity checks
+  // basic checks
```
Comment (dothebart) [inline lib/V8/v8-buffer.cpp:1541]: remove

File: lib/V8/v8-wrapper.h [hunk 14]
```diff
@@ -64,7 +64,7 @@ class V8Wrapper {
   V8Wrapper(v8::Isolate* isolate, STRUCT* object, void (*free)(STRUCT* object),
             v8::Handle<v8::Object> result)
       : _refs(0), _object(object), _free(free), _isolate(isolate) {
-    // sanity checks
+    // basic checks
```
Comment (dothebart) [inline lib/V8/v8-wrapper.h:67]: remove

File: tests/js/client/shell/shell-download.js [hunk 15]
```diff
@@ -78,7 +78,7 @@ function DownloadSuite () {
 ////////////////////////////////////////////////////////////////////////////////
 
     tearDown : function () {
-      // some sanity check as we don't want to unintentionally remove "." or "/"
+      // some basic tests as we don't want to unintentionally remove "." or "/"
```
Comment (dothebart) [inline tests/js/client/shell/shell-download.js:81]: integrity

File: tests/js/client/shell/shell-foxx-paths-spec.js [hunk 16]
```diff
@@ -13,7 +13,7 @@ const baseUrl = origin + '/_db/_system';
 describe('Foxx service path handling', () => {
   const mount = '/unittest/paths';
   before(function () {
-    // Sanity tests
+    // basics tests
```
Comment (dothebart) [inline tests/js/client/shell/shell-foxx-paths-spec.js:16]: remove

File: tests/js/common/shell/shell-fs.js [hunk 17]
```diff
@@ -73,7 +73,7 @@ function FileSystemSuite () {
 ////////////////////////////////////////////////////////////////////////////////
 
     tearDown : function () {
-      // some sanity check as we don't want to unintentionally remove "." or "/"
+      // some basic check as we don't want to unintentionally remove "." or "/"
```
Comment (dothebart) [inline tests/js/common/shell/shell-fs.js:76]: integrity

File: tests/js/server/aql/aql-view-arangosearch-server.js [hunk 18]
```diff
@@ -168,7 +168,7 @@ function iResearchFeatureAqlServerSideTestSuite () {
         docs.push({ _id: "docs/" + docId, _key: docId, "indexField": i }); 
       }
       docsCollection.save(docs);
-      // sanity check. Should be in sync
+      // basic test. Should be in sync
```
Comment (dothebart) [inline tests/js/server/aql/aql-view-arangosearch-server.js:171]: integrity

File: tests/js/server/aql/aql-view-arangosearch-server.js [hunk 19]
```diff
@@ -230,7 +230,7 @@ function iResearchFeatureAqlServerSideTestSuite () {
...
```
Comment (dothebart) [inline tests/js/server/aql/aql-view-arangosearch-server.js:233]: integrity

File: tests/js/server/shell/shell-foxx-legacy-console.js [hunk 20]
```diff
@@ -118,7 +118,7 @@ function ConsoleTestSuite () {
       const max = Date.now();
       console.timeEnd('hi');
       const end = Date.now();
-      expect(max).to.be.greaterThan(min); // sanity checking
+      expect(max).to.be.greaterThan(min); // basic checking...
```
Comment (dothebart) [inline tests/js/server/shell/shell-foxx-legacy-console.js:121]: remove

File: tests/js/server/shell/shell-index.js [hunk 21]
```diff
@@ -108,7 +108,7 @@ function backgroundIndexSuite() {
       // wait for insertion tasks to complete
       waitForTasks();
       
-      // sanity checks
+      // basic checks
```
Comment (dothebart) [inline tests/js/server/shell/shell-index.js:111]: remove

File: arangod/Agency/State.h (hunk 1 above)
Comment (jsteemann) [reply arangod/Agency/State.h:241]: No, it definitely checks for the _existence_ of a collection

File: arangod/RestServer/DatabaseFeature.cpp (hunk 5 above)
Comment (jsteemann) [reply arangod/RestServer/DatabaseFeature.cpp:332]: This is not an integrity check

File: tests/js/common/shell/shell-fs.js (hunk 17 above)
Comment (jsteemann) [reply tests/js/common/shell/shell-fs.js:76]: This is rather a "security" check, so I will adjust it.

File: tests/js/server/shell/shell-index.js (hunk 21 above)
Comment (jsteemann) [reply tests/js/server/shell/shell-index.js]: done

File: tests/js/server/shell/shell-foxx-legacy-console.js (hunk 20 above)
Comment (jsteemann) [reply tests/js/server/shell/shell-foxx-legacy-console.js]: done

**Comment by jsteemann** [comment]
http://172.16.10.101:8080/view/PR/job/arangodb-matrix-pr/10879/

**Comment by jsteemann** [comment]
http://172.16.10.101:8080/view/PR/job/arangodb-matrix-pr/10894/

**Comment by jsteemann** [comment]
http://172.16.10.101:8080/view/PR/job/arangodb-matrix-pr/10897/

**Comment by jsteemann** [comment]
Tests blue

**Pull request merged**